sqlmesh/core/context.py

```diff
@@ -405,7 +405,8 @@ def run(
             latest: The latest time used for non incremental datasets.
             skip_janitor: Whether to skip the jantitor task.
         """
-        self.scheduler(environment=environment or c.PROD).run(start, end, latest)
+        environment = environment or c.PROD
+        self.scheduler(environment=environment).run(environment, start, end, latest)
```

Comment:
why are you passing in environment twice? can you simplify and make it so that you get environment from instance variable?

Reply:
Yeah I missed that. Will fix.

Reply:
Hmm now I am not sure. The environment in the scheduler context is being used to tell the scheduler where to source snapshots from: https://github.com/TobikoData/sqlmesh/blob/330eaaa4d777464f08b8d8542582f067c1f9906d/sqlmesh/core/context.py#L289-L290

The environment we are using here is the environment we are targeting with the run action. So it is possible for example to have a None environment to source snapshots from (there source loaded state) and then an environment to target with run.